src/settings.rs

```diff
@@ -493,6 +527,11 @@ impl Settings {
         self.autokick_bots = kick;
     }
 
+    #[must_use]
+    pub fn web_ui_source(&self) -> UISource {
```

Comment:
Common practice is usually to return a borrow of the struct and allow it to be cloned where it's actually needed rather than cloning it immediately.